dts/bindings/i2c/i2c-controller.yaml

```diff
@@ -9,10 +9,8 @@ bus: i2c
 
 properties:
   "#address-cells":
-    required: true
```

Comment:
Why modifying this file?

This will affect **all** i2c bindings. Unless you have a good reason, don't do. And if you have one, make it its own commit with a self-explanatory commit message.

Reply:
I think this is done because a new child device introduced with DMA settings. Not sure this is right way. How you combine i2-devices and this DMA child device together?

Reply:
Could you check that you can build this?
```
Found devicetree overlay: XXXX/zephyrproject/zephyr/samples/sensor/bme280/boards/intel_rpl_s_crb.overlay
devicetree error: 'reg' property in <Node /pcie0/i2c0/bme280@76 in 'XXXX/zephyrproject/zephyr/misc/empty_file.c'> has length 4, which is not evenly divisible by 12 (= 4*(<#address-cells> (= 2) + <#size-cells> (= 1))). Note that #*-cells properties come either from the parent node or from the controller (in the case of 'interrupts').
```

Reply:
indeed, that's not going to work, merging dma nodes and child.

Reply:
see #60374 for a possible solution